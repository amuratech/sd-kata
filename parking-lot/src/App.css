.App {
  text-align: center;
}

.heading{
  background-color: rgb(158, 84, 177);
  color: white;
  margin-top: 0;
  margin-bottom: 20px;
  padding:3% 0;
  font-size: 3rem;
  position: sticky;
  top:0;
}