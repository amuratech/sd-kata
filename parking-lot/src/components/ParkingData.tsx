import React from 'react'
import '../scss/ParkingDataStyles.css'

import ParkingTable from "./ParkingTable"
import PopUp from './PopUp'
import getAllSlots from '../Services/getAllSlots'
import exitSlot from '../Services/exitSlot'

interface ParkingDataState{
    slotList: Array<Slot>
    showPopUp: boolean
}

class ParkingData extends React.Component<{}, ParkingDataState>{
    constructor(props:any) {
        super(props)
    
        this.state = {
            slotList: Array<Slot>(),
            showPopUp: false 
        }
    }

    toggleGetTicketPopUp = () => {
        this.setState((prevState)=>({
            slotList: prevState.slotList,
            showPopUp: !prevState.showPopUp
        }))
    }

    componentDidMount(){
        this.updateSlots();
    }

    updateSlots = () => {
        getAllSlots()
            .then(res => {
                this.updateTheListWithNewVehicle(res.data);
            }).catch(function(error){
                if (error.response) {
                    console.log(error.response.data);
                    console.log(error.response.status);
                    console.log(error.response.headers);
                    alert("Something went wrong.");
                }
            });
    }

    updateTheListWithNewVehicle = (updateSlotList:Array<Slot>) =>{
        this.setState({
            slotList: updateSlotList
        })
    }

    onCloseModal = () => {
        this.setState((prevState) => ({
            slotList: prevState.slotList,
            showPopUp: false
        }), () => {this.updateSlots()})
    }

    onExit = (vehicleNo:string, slotNo:number) => {
        exitSlot(vehicleNo).then( ()=> 
            this.setState((prevState) => ({
                slotList: prevState.slotList.filter(slot => slot.slotNo != slotNo)
            }))
        ).catch(function(error){
            if (error.response) {
                console.log(error.response.data);
                console.log(error.response.status);
                console.log(error.response.headers);
                alert("Failed to exit ." + error.response.data.message);
            }
        });
    }
    
    render() {
        const {slotList, showPopUp} = this.state;
        return (
            <React.Fragment>
                <PopUp onClose={this.onCloseModal} showPopUp = {showPopUp}/>
                <div>
                    <ParkingTable slotList={slotList} onExit={this.onExit}/>
                    <button className="addNewCarButton"onClick={this.toggleGetTicketPopUp}>Add</button>  
                </div>
            </React.Fragment>
        )
    }
}

export default ParkingData
