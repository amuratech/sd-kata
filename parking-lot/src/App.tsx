import React from 'react';
import './App.css';
import ParkingData from './components/ParkingData';

function App() {
  return (
    <div className="App">
      <h1 className="heading">Parking Lot</h1>
        <ParkingData/>
    </div>
  );
}

export default App;
