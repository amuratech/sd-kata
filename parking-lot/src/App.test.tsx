import React from 'react';
import App from './App';

import {configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ParkingData from './components/ParkingData';
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});

describe("App component testing", () => {
  
  it("should render header of the page", () => {
    const wrapper = shallow(<App />); 
    const header = wrapper.find("h1");
    const text = header.text();
    expect(text).toBe("Parking Lot");
  });

  it("should contain Parking data component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<ParkingData />)).toBe(true);
  })

  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
