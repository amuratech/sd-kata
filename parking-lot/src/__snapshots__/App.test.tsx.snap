// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing renders correctly 1`] = `
<div
  className="App"
>
  <h1
    className="heading"
  >
    Parking Lot
  </h1>
  <div>
    <table
      className="parkingTable"
    >
      <tbody>
        <tr>
          <th>
            Vehicle Number
          </th>
          <th>
            Slot Number
          </th>
          <th>
            Action
          </th>
        </tr>
      </tbody>
    </table>
    <button
      className="addNewCarButton"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
